### PR TITLE
Adds server and certificate fields to SoftwareRecords.

### LINKED ISSUE #247: Track Certificate Expiration and Certificate needs
Notification

### LINKED ISSUE #270: Add QA and DEV Support servers fields to Server Environment Tab

### PR DESCRIPTION
**Pull request by scherztc**
Fixes #270 
Fixes #247 

This PR adds a Required Certificate checkbox to the Software Record.
This PR adds a Certificate Expires date to the Software Record.
These fields can be found on the Server Environment tab.

Create new issues for
- PR to go green under Brakeman

### CODE REVIEW

File: app/views/software_records/_server_environment.html.erb [hunk 1]
```diff
@@ -11,11 +11,23 @@
...
           <% end %>
+          <% unless @software_record.date_cert_expires.to_s.blank? %>
+            <dt class="inline-block">Date that the certificate expires: <span style="color: white; font-weight: normal"><%= @software_record.date_cert_expires %></span></dt>
+          <% end %>
+          <% unless @software_record.monitor_certificates.blank? %>
+            <dt class="inline-block">Uses SSL Certificate:<span style="color: white; font-weight: normal"><%= @software_record.monitor_certificates %></span></dt>
+          <% end %>
           <% unless @software_record.track_uptime.blank? %>
-            <dt class="inline-block">Track uptime:<span style="color: white; font-weight: normal"><%= @software_record.track_uptime %></span></dt>
+            <dt class="inline-block">Tracks uptime wiht Monastic:<span style="color: white; font-weight: normal"><%= @software_record.track_uptime %></span></dt>
```
Comment (crowesn) [inline app/views/software_records/_server_environment.html.erb:30]: Typo 'wiht'

File: app/views/software_records/_form_server_environment.html.erb [hunk 2]
```diff
@@ -18,11 +18,44 @@
...
+              <div class="form-group">
+                <%= form.label :monitor_certificates, 'Track certificates?' %><br/>
+                <%= form.label :monitor_certificates, "Yes", :value => "Yes"  %>
+                <%= form.radio_button :monitor_certificates, "Yes" %>
+                <%= form.label :monitor_certificates, "No", :value => "No" %>
+                <%= form.radio_button :monitor_certificates, "No", :checked => true %>
+              </div>
+            <% else %>
+              <div class="form-group">
+                <%= form.label :monitor_certificates, 'Track uptime?' %><br/>
```
Comment (crowesn) [inline app/views/software_records/_form_server_environment.html.erb:51]: Form label text here should be "Track Certificates". I think this came over from copy/paste?
"Track Uptime" label is duplicated. 

Screen Shot 2023-07-17 at 3 12 35 PM

**Review by crowesn — CHANGES_REQUESTED** [review]
See inline comments with some issues on incorrect label text and a typo. Consider covering these in the specs.

**Review by scherztc — COMMENTED** [review]
@crowesn : The typos and test have been updated and ready for merging.

**Review by crowesn — APPROVED** [review]

**Pull request merged**